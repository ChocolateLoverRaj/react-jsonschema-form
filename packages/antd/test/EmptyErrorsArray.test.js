import React from 'react';
import renderer from 'react-test-renderer';

import '../__mocks__/matchMedia.mock';
import Form from '../src';

const { describe, expect, test } = global;

describe("array fields", () => {
  test("has errors", () => {
    const schema = {
      type: 'object',
        properties: {
          name: {
            type: 'string',
          }
        }
    };
    const tree = renderer
      .create(<Form schema={schema} extraErrors={{ name: { __errors: ["Bad input"] } }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("no errors", () => {
    const schema = {
      type: 'object',
        properties: {
          name: {
            type: 'string',
          }
        }
    };
    const tree = renderer
      .create(<Form schema={schema} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("empty errors array", () => {
    const schema = {
      type: 'object',
        properties: {
          name: {
            type: 'string',
          }
        }
    };
    const tree = renderer
      .create(<Form schema={schema} extraErrors={{ name: { __errors: [] } }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
