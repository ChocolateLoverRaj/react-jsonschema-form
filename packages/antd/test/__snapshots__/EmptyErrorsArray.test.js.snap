// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array fields empty errors array 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group field field-object"
  >
    <fieldset
      id="root"
    >
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      >
        <div
          className="ant-col ant-col-24"
          style={
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <div
            className="form-group field field-string"
          >
            <div
              className="ant-row ant-form-item"
              style={Object {}}
            >
              <div
                className="ant-col ant-col-24 ant-form-item-label"
                style={Object {}}
              >
                <label
                  className=""
                  htmlFor="root_name"
                  title="name"
                >
                  name
                </label>
              </div>
              <div
                className="ant-col ant-col-24 ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input"
                      disabled={false}
                      id="root_name"
                      name="root_name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder=""
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
  </div>
  <div>
    <button
      className="btn btn-info"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`array fields has errors 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="ant-alert ant-alert-error ant-alert-with-description ant-alert-no-icon panel panel-danger errors"
    data-show={true}
  >
    <span
      className="ant-alert-message"
    >
      Errors
    </span>
    <span
      className="ant-alert-description"
    >
      <div
        className="ant-list list-group ant-list-sm ant-list-split"
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
          >
            <li
              className="ant-list-item"
            >
              <div
                className="ant-space ant-space-horizontal"
              >
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginRight": 8,
                    }
                  }
                >
                  <span
                    aria-label="exclamation-circle"
                    className="anticon anticon-exclamation-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="exclamation-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                      <path
                        d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  name: Bad input
                </div>
              </div>
            </li>
          </div>
        </div>
      </div>
    </span>
  </div>
  <div
    className="form-group field field-object"
  >
    <fieldset
      id="root"
    >
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      >
        <div
          className="ant-col ant-col-24"
          style={
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <div
            className="form-group field field-string field-error has-error has-danger"
          >
            <div
              className="ant-row ant-form-item ant-form-item-with-help ant-form-item-has-feedback ant-form-item-has-error"
              style={Object {}}
            >
              <div
                className="ant-col ant-col-24 ant-form-item-label"
                style={Object {}}
              >
                <label
                  className=""
                  htmlFor="root_name"
                  title="name"
                >
                  name
                </label>
              </div>
              <div
                className="ant-col ant-col-24 ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input"
                      disabled={false}
                      id="root_name"
                      name="root_name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder=""
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                      type="text"
                      value=""
                    />
                  </div>
                  <span
                    className="ant-form-item-children-icon"
                  >
                    <span
                      aria-label="close-circle"
                      className="anticon anticon-close-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="close-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                <div
                  className="ant-form-item-explain"
                >
                  <div>
                    <div>
                      Bad input
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
  </div>
  <div>
    <button
      className="btn btn-info"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`array fields no errors 1`] = `
<form
  className="rjsf"
  noValidate={false}
  onSubmit={[Function]}
>
  <div
    className="form-group field field-object"
  >
    <fieldset
      id="root"
    >
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      >
        <div
          className="ant-col ant-col-24"
          style={
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <div
            className="form-group field field-string"
          >
            <div
              className="ant-row ant-form-item"
              style={Object {}}
            >
              <div
                className="ant-col ant-col-24 ant-form-item-label"
                style={Object {}}
              >
                <label
                  className=""
                  htmlFor="root_name"
                  title="name"
                >
                  name
                </label>
              </div>
              <div
                className="ant-col ant-col-24 ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input"
                      disabled={false}
                      id="root_name"
                      name="root_name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder=""
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
  </div>
  <div>
    <button
      className="btn btn-info"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
